src/Workspaces/Core/Portable/Workspace/Host/TextFactory/TextFactoryService.cs

```diff
@@ -17,5 +17,12 @@ public SourceText CreateText(Stream stream, Encoding defaultEncoding, Cancellati
             cancellationToken.ThrowIfCancellationRequested();
             return SourceText.From(stream, defaultEncoding);
         }
+
+        public SourceText CreateText(TextReader reader, Encoding encoding, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            cancellationToken.ThrowIfCancellationRequested();
```

Comment:
@mattwar but that token isn't cancelling the read, it just cancels before we start the read.  Unless I'm missing the calling context.